src/java/org/apache/cassandra/cql3/statements/schema/CopyTableStatement.java

```diff
@@ -229,12 +234,37 @@ public void validate(ClientState state)
         validateDefaultTimeToLive(attrs.asNewTableParams());
     }
 
+    private void cloneIndexesIfNeeded(TableMetadata.Builder builder, TableMetadata sourceTableMeta, KeyspaceMetadata targetKeyspaceMeta)
+    {
+        if (createLikeOption == CreateLikeOption.INDEX)
+        {
+            Indexes.Builder idxBuilder = Indexes.builder();
+            for (IndexMetadata indexMetadata : sourceTableMeta.indexes)
+            {
+                if (indexMetadata.kind == IndexMetadata.Kind.CUSTOM)
+                    throw ire("CUSTOM INDEX is not supported.");
+
+                if (targetKeyspaceMeta.hasIndex(indexMetadata.name))
+                {
+                    throw ire("Target keyspace already has an index with the name : " + indexMetadata.name);
```

Comment:
`Target keyspace ks.tb has already an index with the name ...`

Please add there also keyspace and table